startup.py

```diff
@@ -64,20 +64,65 @@ def scan_software(self, versions=None, display_name=None, icon=None):
 
     def prepare_launch(self, exec_path, args, file_to_open=None):
         """
-        Prepares the given software for launch.
+        Prepares an environment to launch Maya in that will automatically
+        load Toolkit and the tk-maya engine when Maya starts.
 
-        :param str exec_path: Path to DCC executable to launch.
+        :param str exec_path: Path to Maya executable to launch.
         :param str args: Command line arguments as strings.
         :param str file_to_open: (optional) Full path name of a file to open on launch.
         :returns: :class:`LaunchInformation` instance
         """
         required_env = {}
+
+        # Run the engine's userSetup.py file when Maya starts up
+        # by appending it to the env PYTHONPATH.
         startup_path = os.path.join(self.disk_location, "startup")
         sgtk.util.append_path_to_env_var("PYTHONPATH", startup_path)
         required_env["PYTHONPATH"] = os.environ["PYTHONPATH"]
-        required_env["SGTK_ENGINE"] = self.engine_name
-        required_env["SGTK_CONTEXT"] = sgtk.context.serialize(self.context)
+
+        # Check the engine settings to see whether any plugins have been
+        # specified to load.
+        load_plugins = self.get_setting("launch_builtin_plugin") or None
+        if load_plugins:
+            # Parse the specified comma-separated list of plugins
+            find_plugins = [p.strip() for p in load_plugins.split(",") if p.strip()]
```

Comment:
thinking about the support case where someone has trouble configuring this, maybe worth spitting out some debug info around how toolkit splits the string?